### PR TITLE
Fix vmware vm uuid parsing and add DeleteFrom and DetachFrom

### PR DESCRIPTION
**Pull request by harsh-px**
Signed-off-by: Harsh Desai <harsh@portworx.com>

**What this PR does / why we need it**:  

1. User doesn't need to give vsphere vmuuid from environment. It's optional.
2. Add DeleteFrom and DetachFrom API's to detach and delete if invoked from a different node

### CODE REVIEW

**Comment by lpabon** [comment]
FYI: title has `add DeleteFrom and DeleteFrom`

File: pkg/storageops/vsphere/vsphere_util.go [hunk 1]
```diff
@@ -80,10 +80,7 @@ func ReadVSphereConfigFromEnv() (*VSphereConfig, error) {
 		cfg.InsecureFlag = true
 	}
 
-	cfg.VMUUID, err = storageops.GetEnvValueStrict("VSPHERE_VM_UUID")
-	if err != nil {
-		return nil, err
-	}
+	cfg.VMUUID, _ = storageops.GetEnvValueStrict("VSPHERE_VM_UUID")
```
Comment (lpabon) [inline pkg/storageops/vsphere/vsphere_util.go]: curious why this change is needed?

Comment (harsh-px) [reply pkg/storageops/vsphere/vsphere_util.go]: VMUUID is required to be fetched from the env variable`VSPHERE_VM_UUID` only if one is running unit tests for dev.

When running in a real vSphere environment, Portworx (porx) will fetch this from a cloudprovider field that k8s populates in the node object. So we don't need this env. For e.g

```
- apiVersion: v1
  kind: Node
  metadata:
    creationTimestamp: 2018-08-13T03:46:12Z
  spec:
    externalID: 421076c2-e509-7897-e64f-84b0bff79cbe
    providerID: vsphere://421076c2-e509-7897-e64f-84b0bff79cbe
```

**Comment by sangleganesh** [comment]
@harsh-px just for my info, is there a reason this code is a part of libopenstorage ?

**Comment by sangleganesh** [comment]
@harsh-px iirc the reason storage ops of AWS was put in libopenstorage was to have AWS volume driver code reuse the storage ops. gke/vmware don't have that dependency (does not have volume driver implementation for osd). is there any interface/implementation required for osd itself from either gke/vmware ?

**Comment by harsh-px** [comment]
> Please add a UT esp. for DetachFrom for completeness.

@adityadani added

> @harsh-px iirc the reason storage ops of AWS was put in libopenstorage was to have AWS volume driver code reuse the storage ops. gke/vmware don't have that dependency (does not have volume driver implementation for osd). is there any interface/implementation required for osd itself from either gke/vmware ?

Nothing in osd is using gke/vmware. Currently it's only porx. Project rico in libopenstorage was going to use all of these storage ops (https://github.com/libopenstorage/rico/tree/master/pkg/cloudprovider).
